server/src/main/java/com/exacaster/lighter/rest/mappers/StatementCreationResultToApiResponseMapper.java

```diff
@@ -18,6 +31,6 @@ public HttpResponse mapNoSessionExists() {
 
     @Override
     public HttpResponse mapSessionInInvalidState(ApplicationState sessionState) {
-        return HttpResponse.badRequest("invalid session state " + sessionState);
+        return HttpResponse.badRequest(new InvalidSessionStateResponse(sessionState));
```

Comment:
I wonder if it makes sense to somehow implement it with `JsonError` to align with other errors which might be returned by Micronaut service. What are your thoughts on that?

Reply:
i am not so much into `micronaut` and couldn't find anything except for https://guides.micronaut.io/latest/micronaut-error-handling-maven-java.html 
and  https://micronaut-projects.github.io/micronaut-docs-mn1/1.1.3/api/io/micronaut/http/hateoas/JsonError.html i see it is just a general container for a few (optional) properties. If one would like to see why things failed, ie in which state  the session is, he/she would need to parse the message , right ?  
If yes then it is the same what we have now without this PR

Reply:
Bad Request is returned in both cases, when payload validation fails & in this specific case, when session is of invalid state. I think we can jus add additional "message" (or whatever) field that is present on regular micronaut error responses, and put the detais (in this case session state) in some dedicated property.

Reply:
TBH i do not know what you mean... do you mean subclassing/extending `io.micronaut.http.hateoas.JsonError` and in this subclass add additional property (session state), sth like:
```
class InvalidSessionStateError extends JsonError{
    public InvalidSessionStateResponse(ApplicationState applicationState) {
            super("session in invalid state");
            this.sessionState = applicationState;
        }
     public ApplicationState getSessionState() {
            return sessionState;
        }
}

``` 
?

Reply:
I'll merge this & will try to update it, it looks like currently we have some problems with request validations in general.